### PR TITLE
Proactively bump golang.org/x/net to v0.12.0

### PR DESCRIPTION
**Pull request by bertinatto**
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

Proactively bump to `v0.12.0` to avoid `v0.10.0` and `v0.11.0`, which contain a regression added by commit https://github.com/golang/net/commit/82780d606d189ee6d3e9dd4683e493d832f3caaf.

This commit was later reverted in `v0.12.0`: https://github.com/golang/net/commit/4fc2eb99735f9f5a10adbdb9db2727e45bc826c7.

Generated with:

`hack/pin-dependency.sh golang.org/x/net v0.12.0 && hack/update-vendor.sh`

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -20,6 +20,7 @@ require (
 	github.com/armon/circbuf v0.0.0-20150827004946-bbbad097214e
 	github.com/armon/go-socks5 v0.0.0-20160902184237-e75332964ef5
 	github.com/blang/semver/v4 v4.0.0
+	github.com/cilium/ebpf v0.9.1
```
[the remark this answers is not in the document]
Comment (bertinatto) [reply go.mod:23]: Seems to be referenced by cgroups:

```
go mod graph | grep 'github.com/cilium/ebpf@v0.9.1'
k8s.io/kubernetes github.com/cilium/ebpf@v0.9.1
github.com/cilium/ebpf@v0.9.1 github.com/frankban/quicktest@v1.14.0
github.com/cilium/ebpf@v0.9.1 github.com/google/go-cmp@v0.5.6
github.com/cilium/ebpf@v0.9.1 golang.org/x/sys@v0.0.0-20210906170528-6f6e22806c34
github.com/cilium/ebpf@v0.9.1 github.com/kr/pretty@v0.3.0
github.com/cilium/ebpf@v0.9.1 github.com/kr/text@v0.2.0
github.com/cilium/ebpf@v0.9.1 github.com/rogpeppe/go-internal@v1.6.1
github.com/cilium/ebpf@v0.9.1 golang.org/x/xerrors@v0.0.0-20191204190536-9bdfabe68543
github.com/containerd/cgroups@v1.1.0 github.com/cilium/ebpf@v0.9.1
```

This used to be an indirect dependency, see line 152.

[the remark this answers is not in the document]
Comment (liggitt) [reply go.mod:23]: But… why did bumping x/net impact this?

[the remark this answers is not in the document]
Comment (bertinatto) [reply go.mod]: Sorry about that. I had a dirty tree. This should be fixed.

**Pull request merged**